source/bases/Win32Service.c

```diff
@@ -93,6 +93,14 @@ static int Service_SetStatus(udt_ServiceInfo* info, DWORD status)
     serviceStatus.dwControlsAccepted = SERVICE_ACCEPT_STOP;
     if (info->sessionChanges)
         serviceStatus.dwControlsAccepted |= SERVICE_ACCEPT_SESSIONCHANGE;
+    if (exit_code == 0) {
+        serviceStatus.dwWin32ExitCode = 0;
+        serviceStatus.dwServiceSpecificExitCode = 0;
+    }
+    else {
+        serviceStatus.dwWin32ExitCode = ERROR_SERVICE_SPECIFIC_ERROR;
+        serviceStatus.dwServiceSpecificExitCode = exit_code;
+    }
     serviceStatus.dwWin32ExitCode = 0;
```

Comment:
The two lines overwrite the previous value.

Reply:
Thank you, fixed!